### ISSUE TITLE
Build fails with -fno-common

### ISSUE THREAD

**Issue by sbraz** [madrisan/nagios-plugins-linux#48]
Hi, as reported [here](https://bugs.gentoo.org/706536), the plugins fail to build with `-fno-common` in CFLAGS.

It's reproducible with the latest git master: `autoreconf -i && CFLAGS="-O2 -pipe -march=native -fno-common" ./configure && make` will yield:
```
  CCLD     check_clock
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ../lib/libutils.a(progname.o):(.bss+0x8): multiple definition of `program_name'; check_clock.o:(.bss+0x8): first defined here
/usr/lib/gcc/x86_64-pc-linux-gnu/9.2.0/../../../../x86_64-pc-linux-gnu/bin/ld: ../lib/libutils.a(progname.o):(.bss+0x0): multiple definition of `program_name_short'; check_clock.o:(.bss+0x0): first defined here
collect2: error: ld returned 1 exit status
```

**Comment by madrisan** [comment]
GCC documentation:
https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html#index-fno-common-2813

    -fcommon
        [...] pecifies that the compiler places uninitialized global variables in the BSS
        section of the object file. This inhibits the merging of tentative definitions by
        the linker so you get a multiple-definition error if the same variable is
        accidentally defined in more than one compilation unit.

**Comment by sbraz** [comment]
And it's apparently a default in GCC 10:
https://wiki.gentoo.org/wiki/Gcc_10_porting_notes/fno_common

**Comment by madrisan** [comment]
Ok. Thanks for the report!
The following patch should fix this issue:
```
diff --git a/include/progname.h b/include/progname.h
index 576ed16..847122a 100644
--- a/include/progname.h
+++ b/include/progname.h
@@ -25,10 +25,10 @@ extern "C" {
 #endif
 
   /* String containing name the program is called with.  */
-  const char *program_name;
+  extern const char *program_name;
 
   /* String containing a short version of 'program_name'.  */
-  const char *program_name_short;
+  extern const char *program_name_short;
 
   /* Set program_name, based on argv[0].
      argv0 must be a string allocated with indefinite extent, and must not be
```

**Comment by sbraz** [comment]
Thanks Davide, that works. Will you make a new release soon or should I backport the patch?

**Comment by madrisan** [comment]
I think the changes in the project are not enough to require a new version.
Unless gcc 10 will be released soon.
If you need a quick fix, you should backport the patch.

**Comment by sbraz** [comment]
> If all goes well, GCC 10.1 will be the first stable release of GCC 10 and should be out in March or April of 2020.

https://www.phoronix.com/scan.php?page=news_item&px=GCC-10-Enters-Stage-Three

I'll backport it now then. I assume you'll add this to master anyway?

**Comment by madrisan** [comment]
Yes of course. The patch will be in the master branch in the evening.

**Comment by madrisan** [comment]
Fixed by the pull request [#49](https://github.com/madrisan/nagios-plugins-linux/pull/49).
